src/l1_fetcher.rs

```diff
@@ -0,0 +1,123 @@
+use ethers::{abi::Contract, prelude::*, providers::Provider};
+use eyre::Result;
+use state_reconstruct::parse_calldata;
+use state_reconstruct::CommitBlockInfoV1;
+use state_reconstruct::BLOCK_STEP;
+use state_reconstruct::GENESIS_BLOCK;
+use state_reconstruct::ZK_SYNC_ADDR;
```

Comment:
These can be merged; there might be some `rustfmt.toml` setting that automatically assures this, which we should investigate.